tnnt/hardfought_utils.py

```diff
@@ -0,0 +1,79 @@
+from django.contrib.auth.backends import BaseBackend
+from django.contrib.auth.models import User
+from scoreboard.models import Player
+from .settings import DGL_DATABASE_PATH
+import sqlite3
+import crypt
+
+def get_dgl_cursor():
+    # open in read-only mode
+    dgl_conn = sqlite3.connect('file:' + DGL_DATABASE_PATH + '?mode=ro', uri=True)
+    dgl_conn.row_factory = sqlite3.Row
+    return dgl_conn.cursor()
+
+# Function that looks up a player in first the Player table, then if not found
+# there then in dgamelaunch database. If found in dgamelaunch, create the player
+# in the Player table and return
+def find_player(findname):
+    try:
+        return Player.objects.get(name=findname)
+    except Player.DoesNotExist as e:
+        try:
+            dgl_curs = get_dgl_cursor()
+            uname = dgl_curs.execute('SELECT username FROM dglusers WHERE username = ?',
+                                     (findname,)).fetchone()
+        except sqlite3.Error as sqlite_e:
+            print('ERROR connecting to or executing SQL on sqlite database')
+            print('  db path:', DGL_DATABASE_PATH)
+            print('  username:', username)
+        if uname is None:
```

Comment:
This behaviour will mean I need a dummy dgl database for testing that has one each of everyone's nick...
I suppose I can generate such a thing in a similar way to your alice, bob, carol, dan one?